src/test/kotlin/uk/gov/justice/digital/hmpps/prisonregister/integration/GetPrisonEmailResourceTest.kt

```diff
@@ -0,0 +1,223 @@
+    assertEquals("Could not find email address for BRI and social-visit.", bodyText)
+  }
+
+  @Test
+  fun `When purpose type does not exist, then appropriate error is show`() {
+    // Given
+    val prisonId = "BRI"
+    val endPoint = "/secure/prisons/id/$prisonId/type/i-do-not-exist/email-address"
+
+    // When
+    val responseSpec = doStartAction(endPoint, createAnyRole())
+
+    // Then
+    responseSpec.expectStatus()
+      .isBadRequest
+
+    val bodyText = getResponseBodyText(responseSpec)
+    assertEquals("Value for ContactPurposeType is not of a know type i-do-not-exist.", bodyText)
+  }
+
+  @Test
+  fun `When no role is give to get email for videolink-conferencing-centre, status unauthorized is returned`() {
+    // Given
+    val prisonId = "BRI"
+    val endPoint = "/secure/prisons/id/$prisonId/$VCC/email-address"
+
+    // When
+    val responseSpec = doStartActionNoRole(endPoint)
+
+    // Then
+    responseSpec.expectStatus()
+      .isUnauthorized
+  }
+
+  @Test
+  fun `When no role is give to get email for offender-management-unit, status unauthorized is returned`() {
+    // Given
+    val prisonId = "BRI"
+    val endPoint = "/secure/prisons/id/$prisonId/$OMU/email-address"
+
+    // When
+    val responseSpec = doStartActionNoRole(endPoint)
+
+    // Then
+    responseSpec.expectStatus()
+      .isUnauthorized
+  }
+
+  @Test
+  fun `When no role is give to get email for given type, status unauthorized is returned`() {
+    // Given
+    val prisonId = "BRI"
+    val contactPurposeType = ContactPurposeType.SOCIAL_VISIT
+    val endPoint = "/secure/prisons/id/$prisonId/type/${contactPurposeType.value}/email-address"
+
+    // When
+    val responseSpec = doStartActionNoRole(endPoint)
+
+    // Then
+    responseSpec.expectStatus()
+      .isUnauthorized
+  }
+
+  private fun doStartActionNoRole(endPoint: String): ResponseSpec {
+    return webTestClient.get()
+      .uri(endPoint).exchange()
+  }
+
+  private fun doStartAction(endPoint: String, headers: (HttpHeaders) -> Unit): ResponseSpec {
+    return webTestClient.get()
+      .uri(endPoint)
+      .headers(headers)
+      .exchange()
+  }
+
+  private fun createAnyRole(): (HttpHeaders) -> Unit = setAuthorisation(roles = listOf("ANY_ROLE"), scopes = listOf("something"))
+
+  private fun getResponseBodyText(responseSpec: ResponseSpec): String {
+    return String(responseSpec.expectBody().returnResult().responseBody, StandardCharsets.UTF_8)
+  }
+
+  private fun createDBData(prisonId: String, contactPurposeType: ContactPurposeType): Prison {
+    val prison = Prison(prisonId, "$prisonId Prison", active = true)
+    prisonRepository.save(prison)
+
+    val emailAddress = emailAddressRepository.save(EmailAddress("aled@moj.gov.uk"))
+    contactDetailsRepository.saveAndFlush(
+      ContactDetails(
+        prison.prisonId,
+        prison,
+        contactPurposeType,
+        emailAddress,
+      ),
+    )
+
+    return prison
+  }
```

Comment:
do we need a test for - delete where email address does not exist and for create where email address already exists. Ignore if already there.

Reply:
The create question you asked was covered by this test :
`When an one email is added for more than one prison, only one email is persisted`